lib/recommendation-server.js

```diff
@@ -82,17 +89,36 @@ RecommendationServer.prototype = {
       req.send();
     }
   },
+
+  getResultClass: function(data) {
+    let types = Object.keys(data.enhancements);
+    if (types.indexOf('tld') > -1) {
+      return this.resultClasses.tld;
+    } else if (types.indexOf('wikipedia') > -1) {
+      return this.resultClasses.wikipedia;
+    }
+    return null;
+  },
+
   search: function(payload) {
+    let server = this;
```

Comment:
In this repo, I've been systematically binding the value of `this` using two steps: 
- `bind()` the `this` value inside the constructor, if the function is called externally without the object reference (like event handler callbacks, setTimeout callbacks, or xhr success callbacks)
- use arrow fns for inner functions that can be anonymous (callbacks that never need to be detached by reference later)

It looks like `this.search` is bound to `this` in the constructor, so `this._search()` should be a valid reference in here. And the inner function is an arrow func, so the `this` binding should work there, too. Maybe give it another try using `this` in place of `server`?